test/test_shape_ops.py

```diff
@@ -631,7 +631,7 @@ def gen_nontrivial_input(shape, dtype, device):
                     "scalar type Long",
                     lambda: torch.nonzero(tensor, out=torch.empty([], dtype=torch.float, device=device))
                 )
-            if self.device_type == 'cuda':
+            if self.device_type == TEST_DEVICE_TYPE:
```

Comment:
If I read correct, this checks that if out= is not on the same device, this fails.
So this could be changed to `self.device_type != "cpu"` ? That sounds like a simpler changer?

Reply:
> If I read correct, this checks that if out= is not on the same device, this fails. So this could be changed to `self.device_type != "cpu"` ? That sounds like a simpler changer?

I don't just want to change the file 'test_shape_ops.py'. I mainly want to add the variable TEST_DEVICE_TYPE in 'common_utils.py'. The change in the 'test_shape_ops.py' is an example that I want to show how it can be used. 
I think it can be used in more test files, then it will be more convenient for users to adapt tests on custom device.

Reply:
But this variable is kind of a lie right now? It can only be cuda or privateuse1 in your implementation. So I'm not 100% buying the argument that this is a general variable saying "what is the device being tested right now"

Reply:
> But this variable is kind of a lie right now? It can only be cuda or privateuse1 in your implementation. So I'm not 100% buying the argument that this is a general variable saying "what is the device being tested right now"

You are right.